dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -947,7 +947,7 @@ private String getNullTerminatedString(HpccSrcType stype) throws IOException
 
                 for (int j = 0; j < readSize-1; j += 2)
                 {
-                    if (scratchBuffer[j] == '\0' && scratchBuffer[j + 1] == '\0')
+                    if (scratchBuffer[strByteLen + j] == '\0' && scratchBuffer[strByteLen + j + 1] == '\0')
```

Comment:
is readIntoScratchBuffer only writing into scratchBuffer[strByteLen ...strByteLen+readSize]?